pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -206,6 +207,51 @@ func (n *nodeHealthMap) set(name string, data *nodeHealthData) {
 	n.nodeHealths[name] = data
 }
 
+type podsToEvictMap struct {
+	lock sync.Mutex
+	//podMap: nodeName -> podUID -> pod
+	podMap map[string]map[string]*v1.Pod
+}
+
+func newPodsToEvictMap() *podsToEvictMap {
+	return &podsToEvictMap{
+		podMap: make(map[string]map[string]*v1.Pod),
+	}
+}
+
+func (p *podsToEvictMap) add(node string, pods []v1.Pod) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	if _, exists := p.podMap[node]; !exists {
+		p.podMap[node] = make(map[string]*v1.Pod)
+	}
+	for i := range pods {
+		p.podMap[node][string(pods[i].UID)] = &pods[i]
```

Comment:
what's the point of casting? why can't we make map[types.UID] instead of map[string] ?